### PR TITLE
gtklock-*-module: init at 2.0.{0,1}

### PR DESCRIPTION
**Pull request by Aleksanaa**
###### Description of changes

Added three gtklock modules written by its main dev jovanlanik. All fully tested. (gtklock seems to block the PrtSc key)

IMG20230419235618

The author suggests that these plugins should be matched with the corresponding version of gtklock to avoid potential issues. This is something to consider when updating gtklock.

also ping gtklock package maintainer @dit7ya 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/wayland/gtklock/playerctl-module.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  version = "2.0.1";
+
+  src = fetchFromGitHub {
+    owner = "jovanlanik";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-kzGgqFDTeKL6Pfjram7pqVcIm8Avxsvpn1qFrcpd8dw=";
+  };
+
+  nativeBuildInputs = [ gnumake pkg-config ];
```
Comment (wegank) [inline pkgs/tools/wayland/gtklock/playerctl-module.nix:22]: `gnumake` being part of `stdenv`, this shouldn't be necessary.

```suggestion
  nativeBuildInputs = [ pkg-config ];
```

**Comment by wegank** [comment]
@ofborg build gtklock-playerctl-module gtklock-powerbar-module gtklock-userinfo-module

**Review by wegank — APPROVED** [review]

**Review by NickCao — APPROVED** [review]
diff LGTM, please split the packages into individual commits.

**Comment by Aleksanaa** [comment]
Done.

**Pull request merged**